### PR TITLE
Added hide/show node buttons to Network View

### PR DESCRIPTION
**Pull request by MichaelWrana**
Refs #833 
Current Implementation:

1. User presses shift+left-click+drag to select an area on screen, and nodes within that area are highlighted:

image

2. User can then press the struck-through eye icon to hide those selected nodes:

image
image

3. Steps 1 & 2 can be repeated until no nodes remain in the graph.

4. Pressing the 'eye' icon will re-show any nodes that have been hidden previously, reproducing the original graph.

### CODE REVIEW

**Comment by MichaelWrana** [comment]
Need to fix edges not reappearing issue, don't pull for now.

**Comment by jvwong** [comment]
Offline discussion with @jvwong and @MichaelWrana: This feature brings to forefront several bugs that impact function. We will list, triage in separate issues.

File: src/client/common/components/base-network-view/config/index.js [hunk 1]
```diff
@@ -21,6 +21,35 @@ const resetToDefaultLayout = (props) => {
...
 
+//This list of nodes is created from shift+drag box select.  See box-select.js
+const hideSelectedNodes = (props) => {
+  const cy = props.cy;
+  let nodesToHide = cy.selectedNodesToHide;
+  for(let i in nodesToHide){
+    let node = nodesToHide[i];
+    node.hide();
+  }
+  cy.selectedNodesToHide = [];
```
Comment (maxkfranz) [inline src/client/common/components/base-network-view/config/index.js]: This is the relevant line.  Shouldn't be setting things in `cy`...

File: src/client/common/components/base-network-view/config/index.js [hunk 2]
```diff
@@ -21,6 +21,35 @@ const resetToDefaultLayout = (props) => {
   cy.layout(props.layoutConfig.defaultLayout.options).run();
 };
 
+//This list of nodes is created from shift+drag box select.  See box-select.js
+const hideSelectedNodes = (props) => {
+  const cy = props.cy;
+  let nodesToHide = cy.selectedNodesToHide;
+  for(let i in nodesToHide){
+    let node = nodesToHide[i];
+    node.hide();
```
Comment (maxkfranz) [inline src/client/common/components/base-network-view/config/index.js]: Use classes.  Do not use `.show()` or `.hide()`

**Comment by maxkfranz** [comment]
show() and hide() won't allow for composition with other show/hide features like the expand/collapse button.  It makes things fragile.  It's better to use classes so your stylesheet can handle the overlapping cases in one place

**Comment by MichaelWrana** [comment]
This won't work with the new pathway viewer changes

**Pull request closed without merging**